fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkimport/ImportJobListener.java

```diff
@@ -8,29 +8,46 @@
 
 import java.util.HashMap;
 import java.util.List;
+import java.util.logging.Logger;
 
 import javax.batch.api.listener.JobListener;
+import javax.batch.operations.JobOperator;
+import javax.batch.runtime.BatchRuntime;
+import javax.batch.runtime.JobExecution;
 import javax.batch.runtime.context.JobContext;
 import javax.inject.Inject;
 
 public class ImportJobListener implements JobListener {
+    private static final Logger logger = Logger.getLogger(ImportJobListener.class.getName());
     @Inject
     JobContext jobContext;
 
-    private long jobStartTimeInMS, jobEndTimeInMS;
-
     public ImportJobListener() {
 
     }
 
-    @SuppressWarnings("unchecked")
+
+    @SuppressWarnings({"unchecked" })
     @Override
     public void afterJob() {
+        // jobExecution.getEndTime() for current execution always returns null, so we use system current time as the end time for current execution.
+        long currentExecutionEndTimeInMS = System.currentTimeMillis();;
+
         // Used for generating response for all the import data resources.
         List<ImportCheckPointData> partitionSummaries = (List<ImportCheckPointData>)jobContext.getTransientUserData();
         // Used for generating performance measurement per each resource type.
         HashMap<String, ImportCheckPointData> importedResourceTypeSummaries = new HashMap<>();
 
+        JobOperator jobOperator = BatchRuntime.getJobOperator();
```

Comment:
it's cool to see this works in the context of the webapp (I'm sure this simplifies things)

Reply:
yes, totally agree